components/admin_console/schema_admin_settings.jsx

```diff
@@ -99,6 +102,13 @@ export default class SchemaAdminSettings extends React.Component {
     handleSubmit = async (e) => {
         e.preventDefault();
 
+        if (this.state.confirmNeeded) {
+            this.setState({
+                showConfirm: this.state.confirmNeeded,
```

Comment:
Could we set this directly to true? Also, do we need a `showConfirm` ? I only feel it's redundant because if `confirmNeeded` is true, then we probably need to `showConfirm` too. Thoughts?
```suggestion
                showConfirm: true,
```

Reply:
- `showConfirm` needs to be set _after_ the user clicks on the save button 
- `confirmNeeded` will be set any time the value of the setting changes

Technically `showConfirm` could be a boolean but then I would have to pass
`showConfirm={this.state.showConfirm && this.state.confirmNeeded === setting.key}` 

as the prop to the custom component instead of
`showConfirm={this.state.showConfirmId === setting.key}`